scrape/scrape.go

```diff
@@ -154,6 +154,11 @@ func newScrapePool(cfg *config.ScrapeConfig, app storage.Appendable, offsetSeed
 		return nil, err
 	}
 
+	switch cfg.MetricNameValidationScheme {
+	case model.LegacyValidation, model.UTF8Validation:
+	default:
+		return nil, errors.New("newScrapePool: MetricNameValidationScheme must be set in scrape configuration")
+	}
```

Comment:
here's that pattern again.  maybe we just need a validationScheme.Validate() call?

Reply:
See [other comment](https://github.com/prometheus/prometheus/pull/16928#discussion_r2267167887).